protos/sources/vega/data_source.proto

```diff
@@ -54,6 +55,41 @@ message DataSourceSpecConfiguration {
   repeated vega.data.v1.Filter filters = 2;
 }
 
+// DataSourceSpecEthCall specifies a data source that derives its content from calling
+// a read method on an ethereum contract.
+message DataSourceSpecEthCall {
+  // Ethereum address of contract to call
+  string address = 1;
+  // The ABI of that contract
+  google.protobuf.ListValue  abi = 2;
+  // Name of the method on the contract to call.
+  string method = 3;
+  // List of arguments to pass to method call.
+  // The protobuf 'Value' wraps an arbitaray JSON type, which is mapped to an ethereum type according to the ABI.
+  repeated google.protobuf.Value args = 4;
+  // Method for determining when to call the contract, and (optionally) at what interval
+  EthCallTrigger trigger = 5;  
+}
+
+// EthCallTrigger determines when the contract method should be called. 
+message EthCallTrigger {
+  oneof trigger{
+    EthTimeTrigger time_trigger = 1;
+  }
+}
+
+// EthTimeTrigger is a trigger for an ethereum call based on the ethereum block timestamp. Can be one-off or repeating.
+message EthTimeTrigger {
+  // The first block in which ethereum time is greater or equal to this time (in seconds since the UNIX epoch)
+  // if not specified, begin repeating straight away.
+  optional uint64 initial = 1;  
+  // Optionally repeat the call every n seconds
+  optional uint64 every = 2;
+  // Optionally stop repeating the call once etherem time is greater than this time (in seconds since the UNIX epoch)
```

Comment:
```suggestion
//  Trigger for an Ethereum call based on the Ethereum block timestamp. Can be one-off or repeating.
message EthTimeTrigger {
  // First block in which Ethereum time is greater or equal to this time, in seconds.
  // If not specified, begin repeating straight away
  optional uint64 initial = 1;  
  // Optionally repeat the call every n seconds
  optional uint64 every = 2;
  // Optionally stop repeating the call once Ethereum time is greater than this time, in seconds
```

Reply:
I think the UNIX epoch part is important - you can't measure absolute time in seconds without referencing an epoch. You could say something like 'encoded as a UNIX timestamp' but that requires you to know what that means

Reply:
I've edited my comment to say "in Unix seconds", which is the format we use in other doc-strings

Reply:
Ok thanks - I'll fold all those suggestions in